server.js

```diff
@@ -0,0 +1,136 @@
+var express = require('express');
+var app = express();
+app.use(express.static('public'));
+var http = require('http').Server(app);
+//var io = require('socket.io')(http);
+
+var bodyParser = require('body-parser');
+app.use( bodyParser.json() );       // to support JSON-encoded bodies
+app.use(bodyParser.urlencoded({     // to support URL-encoded bodies
+	extended: true
+}));
+
+var heroesRec = require('./heroes');
+var heroes = new heroesRec.Heroes();
+
+
+app.get('/', function(req, res){
+	res.sendFile(__dirname + '/public/view/index.html');
+});
+
+app.route('/heroes')
+	.get( function (req, res)
+	{
+		res.send(heroes.getHeroes());
+	})
+	.post( function (req, res)
+	{
+		const name = req.body.name;
+		const id =  req.body.id;
+
+		if(id === null)
```

Comment:
if(!id) is better,, its check if id exist(not null and not undefined).
in this case you will get undefined as id.